packages/eslint-plugin/utils.js

```diff
@@ -100,14 +101,29 @@ function isImported( node ) {
 	return false;
 }
 
+const hasFunctionType = ( type ) => {
+	return [
+		'ArrowFunctionExpression',
+		'FunctionDeclaration',
+		'FunctionExpression',
+	].includes( type );
+};
+
 function isFunction( node ) {
 	if ( ! node ) {
 		return false;
 	}
 
-	const isFunctionDeclaration = node.type && (
-		[ 'ArrowFunctionExpression', 'FunctionDeclaration', 'FunctionExpression' ].includes( node.type )
-	);
+	if (
+		node?.type === 'Identifier' && (
+			hasFunctionType( node?.parent?.type ) ||
+			hasFunctionType( node?.parent?.init?.type )
+		)
+	) {
+		return true;
+	}
+
+	const isFunctionDeclaration = hasFunctionType( node?.type );
```

Comment:
I think we weren't using optional chaining in these files before because our version of node was too old, but now that we require Node 14 via NVM this should be fine. 👍🏻